src/topp/MapAlignerTreeGuided.cpp

```diff
@@ -300,14 +300,15 @@ void TOPPMapAlignerTreeGuided::getPeptideSequences_(const vector<PeptideIdentifi
 void TOPPMapAlignerTreeGuided::extractSeqAndRt_(const vector<FeatureMap>& feature_maps,
         vector<SeqAndRTList>& maps_seq_and_rt, vector<vector<double>>& maps_ranges)
 {
+  vector<double> rts_tmp;
```

Comment:
looking at this again, you could just pass `maps_ranges[i]` to `addPeptideSequences_()` like you do with the `maps_seq_and_rt[i]` and omit the temporary vector.
The `reserve` still makes sense probably, but then also call `.shrink_to_fit()` instead of `clear()`, just to make sure we are not wasting big amounts of memory.

Reply:
`shrink_to_fit` I did not know at all. But after using `push_back()`  shouldn't you always use `shrink_to_fit` to adjust the capacity, then? 
I would leave out `reserve` now, unless you recommend it. But then I would have to use it in several other places as well?!

Reply:
`shrink_to_fit` only makes sense if you called `reserve` and you know that you overestimated capacity (i.e. the final size is smaller).
The only way to know if reserve is worth it (if you know upfront the capacity will be larger than size!), is to benchmark.

If you already know the exact number of items upfront, reserve or resize definitely make sense. you can `reserve + push_back` or `resize + [i]`... the latter might be faster (for less obvious reasons). No shrink required in both cases.

Reply:
shrink_to_fit reallocates the memory of the vector and needs to internally copy all data to the new memory location (http://www.cplusplus.com/reference/vector/vector/shrink_to_fit/). For large vectors it might make sense. Shrinking many small ones and reallocating them might lead to more memory fragmentation so I think it is generally not advisable to shrink to fit every vector.

Reply:
no, `shrink_to_fit` CAN realloc, but may very well do nothing (e.g. if the capacity is below the minimum block size that the allocator would use anyway).
You just have to trust the implementation to do 'sensible' things.